### ISSUE TITLE
Failing testing of datalad "core"

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad-next#705]
e.g. https://github.com/datalad/datalad/actions/runs/9163271335/job/25191987511#step:12:2179

```
patches/tests/test_push.py::test_nested_pushclone_cycle_allplatforms create(ok): /tmp/datalad_temp_test_nested_pushclone_cycle_allplatformsyh646c0g/super (dataset)
Warning:  Failed to import vcr, no cassettes will be available: AttributeError(module 'urllib3.connectionpool' has no attribute 'VerifiedHTTPSConnection') 
add(ok): sub (dataset)
add(ok): .gitmodules (file)
save(ok): . (dataset)
create(ok): sub (dataset)
action summary:
  add (ok: 2)
  create (ok: 1)
  save (ok: 1)
add(ok): file2.txt (file)
save(ok): sub (dataset)
add(ok): sub (dataset)
add(ok): .gitmodules (file)
add(ok): file1.txt (file)
save(ok): . (dataset)
action summary:
  add (ok: 4)
  save (ok: 2)
[INFO] Start checking pre-existing sibling configuration Dataset(/tmp/datalad_temp_test_nested_pushclone_cycle_allplatformsyh646c0g/super) 
Warning:  Failed to import vcr, no cassettes will be available: AttributeError(module 'urllib3.connectionpool' has no attribute 'VerifiedHTTPSConnection') 
[INFO] Finished checking pre-existing sibling configuration Dataset(/tmp/datalad_temp_test_nested_pushclone_cycle_allplatformsyh646c0g/super) 
[INFO] Creating a new RIA store at /tmp/datalad_temp_test_nested_pushclone_cycle_allplatformsnyqfdzre 
Error:  AssertionError 
....
=========================== short test summary info ============================
FAILED patches/tests/test_push.py::test_nested_pushclone_cycle_allplatforms - datalad.runner.exception.CommandError: CommandError: 'datalad create-sibling-ria --recursive -s store ria+file:///tmp/datalad_temp_test_nested_pushclone_cycle_allplatformsnyqfdzre --new-store-ok' failed with exitcode 1
====== 1 failed, 409 passed, 30 skipped, 17 warnings in 790.82s (0:13:10) ======
```

**Comment by mih** [comment]
This should be fixed in 1.4.1 and with https://github.com/datalad/datalad-next/pull/696 in particular.

**Comment by christian-monch** [comment]
The problem seems to be that the datalad-next patches are partially used and partially not used.


The class `CreateSiblingRia` stems from datalad-core, i.e. from `datalad.distributed.create_sibling_ria`. But the function `_ensure_version` in `ria_utils` is the function `ria_utils__ensure_version` from a datalad-next patch, i.e. from `datalad_next.patches.ria_utils`.

Running tests with log-level 8 reveals this fact in the assert message. 

```
[Level 8] AssertionError [main.py:_run_with_exception_handler:190,exec.py:call_from_parser:107,base.py:_execute_command_:873,utils.py:_process_results:319,create_sibling_ria.py:__call__:428,ria_utils.py:create_store:177,ria_utils.py:ria_utils__ensure_version:48] 
```

The test passes for me if the environment variable `DATALAD_EXTENSIONS_LOAD` is set to `next`.  That should be done, when testing the datalad-next-extension.

I opened PR [datalad #7605](https://github.com/datalad/datalad/pull/7605), which sets `DATALAD_EXTENSIONS_LOAD=next` in the datalad-next extension tests
